File: src/test/java/com/github/kayjamlang/tests/containers/ContainerEmptyTest.java
package com.github.kayjamlang.tests.containers;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.containers.Container;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class ContainerEmptyTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("{}"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(Container.class, expression.getClass());

        Container container = (Container) expression;
        assertEquals(0, container.children.size());
    }
}

File: src/test/scala/com/github/kayjamlang/tests/containers/ContainerEmptyTest.scala
package com.github.kayjamlang.tests.containers

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.containers.Container
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object ContainerEmptyTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("{}"))
  }
}

class ContainerEmptyTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = ContainerEmptyTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[Container], expression getClass)
    val container = expression.asInstanceOf[Container]
    assertEquals(0, container.children size)
  }
}